src/plugins/data_views/server/ui_settings.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { schema } from '@kbn/config-schema';
+import { DEFAULT_FIELD_CACHE_FRESHNESS } from './constants';
+
+export const cacheMaxAge = {
+  'data_views:cache_max_age': {
+    name: i18n.translate('dataViews.advancedSettings.cacheMaxAgeTitle', {
+      defaultMessage: 'Field cache max age (in seconds)',
+    }),
+    value: DEFAULT_FIELD_CACHE_FRESHNESS,
+    description: i18n.translate('dataViews.advancedSettings.cacheMaxAgeText', {
+      defaultMessage:
+        "Sets the 'max-age' cache header value for data view fields API requests. A value of 0 will disable caching. A hard reload of Kibana may be necessary for your browser to pick up the new setting.",
```

Comment:
It may be worth getting some input from a writer on this. I'm not sure the current description makes it clear enough that modifying this value may have no effect until each user either clears their browser cache or their current cached value expires.

Reply:
@amyjtechwriter Could you provide some advice here? This text is displayed in Advanced settings.

Reply:
I propose changing this message to:
'Sets how long data view fields API requests are cached in seconds. A value of 0 turns off caching. Modifying this value may not take immediate effect, users need to clear browser cache or wait until the current cache expires. To see immediate changes, try a hard reload of Kibana.'

It's not shorter but hopefully clearer. What do you think @davismcphee and @mattkime?

Reply:
Should we add information here, that when user's role are changing, the cache is not invalidated automatically? ... or something like this?

Reply:
@kertal @davismcphee I'd rather explain the relationship between the cache and user's role in kibana docs instead of next to a setting. Not a strong preference though, what do you think?